src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -224,6 +215,27 @@ private TestPackage CreateTestPackage(string assemblyName)
             return package;
         }
 
+        /// <summary>
+        /// Sets test parameters, handling backwards compatibility.
+        /// </summary>
+        private static void SetTestParameters(IDictionary<string, object> runSettings, IDictionary<string, string> testParameters)
+        {
+            runSettings[PackageSettings.TestParametersDictionary] = testParameters;
+
+            if (testParameters.Count != 0)
+            {
+                // Kept for backwards compatibility with old frameworks.
+                // Reserializes the way old frameworks understand, even if the parsing above is changed.
+                // This reserialization cannot be changed without breaking compatibility with old frameworks.
+
+                var oldFrameworkSerializedParameters = new StringBuilder();
+                foreach (var parameter in testParameters)
+                    oldFrameworkSerializedParameters.Append(parameter.Key).Append('=').Append(parameter.Value).Append(';');
+
```

Comment:
OK then... it's one of those matters of individual style I keep preaching about. :smile: